src/ol/collection.js

```diff
@@ -225,6 +249,20 @@ ol.Collection.prototype.updateLength_ = function() {
 
 
 /**
+ * @private
+ * @param {T} elem Element.
+ * @param {number=} opt_except Optional index to ignore.
+ */
+ol.Collection.prototype.assertUnique_ = function(elem, opt_except) {
+  for (var i = 0, ii = this.array_.length; i < ii; ++i) {
+    if (this.array_[i] === elem && i !== opt_except) {
+      throw new Error('Duplicate item added to a unique collection');
```

Comment:
> I'm compelled by the byte saving argument. But if we are really committed to that, we'd get rid of ol.Collection entirely.

Also true. So I think you can merge this, and removal of `ol.Collection` is something we can save for later.